lib/liquid/filters.ex

```diff
@@ -478,35 +482,36 @@ defmodule Liquid.Filters do
   @doc """
   Recursively pass through all of the input filters applying them
   """
+  @spec filter(list(), String.t()) :: String.t() | list()
   def filter([], value), do: value
 
   def filter([filter | rest], value) do
     [name, args] = filter
 
     args =
       for arg <- args do
-        Liquid.quote_matcher() |> Regex.replace(arg, "")
+        Regex.replace(Liquid.quote_matcher(), arg, "")
       end
 
-    functions = Functions.__info__(:functions)
+    functions = @filters_modules |> Enum.map(&set_module/1) |> List.flatten()
     custom_filters = Application.get_env(:liquid, :custom_filters)
 
     ret =
-      case {name, functions[name], custom_filters[name]} do
+      case {name, custom_filters[name], functions[name]} do
         # pass value in case of no filters
         {nil, _, _} ->
           value
 
-        # pass non-existend filter
+        # pass non-existent filter
         {_, nil, nil} ->
           value
 
-        # Fallback to custom if no standard
+        # Fallback to standard if no custom
         {_, nil, _} ->
```

Comment:
since both the `name` and `function` are used in the body of this function, it would be nicer and more explicit to match on `{name, nil, function} ->`
This way you can change
`apply_function(functions[name], name, [value | args])`
to
`apply_function(function, name, [value | args])`

Reply:
Done